external/kafka-0-10-token-provider/src/test/scala/org/apache/spark/kafka010/KafkaConfigUpdaterSuite.scala

```diff
@@ -76,6 +79,25 @@ class KafkaConfigUpdaterSuite extends SparkFunSuite with KafkaDelegationTokenTes
     val params = Map(
       CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG -> bootStrapServers
     )
+    testWithTokenSetValues(params, updatedParams => {
+      assert(updatedParams.get(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG) === SASL_SSL.name)
+    })
+  }
+
+  test("setAuthenticationConfigIfNeeded with token should set values and override protocol") {
```

Comment:
Restructured a bit. Let me know if you've better idea and we can adapt it.

Reply:
I guess `setAuthenticationConfigIfNeeded` is not important here - we're verifying that protocol in source/sink takes effect (preference) if protocol is specified for both source/sink and delegation token. It would be ideal if we could represent the point.

Reply:
The way how I'm structuring tests is: given -> when -> then.
`setAuthenticationConfigIfNeeded` is the action (when part) which executed. Maybe we can rephrase it like: `setAuthenticationConfigIfNeeded with token should prefer overridden protocol`?

Reply:
"overriden" <- this brings confusion. Without knowing details, we don't know what's the basic configuration for protocol and what will override it. The thing is simple, once we specify protocol in source/sink, it will always take effect (regardless of protocol in token).

Reply:
See your point: `setAuthenticationConfigIfNeeded with token should prefer source protocol`?

Reply:
New test name looks OK to me.

Reply:
I've taken over what Marcelo suggested which is more or less the same in a negated way.